src/Build/BuildCheck/Infrastructure/BuildAnalyzerConfigurationInternal.cs

```diff
@@ -10,25 +11,31 @@ namespace Microsoft.Build.Experimental.BuildCheck.Infrastructure;
 /// </summary>
 internal sealed class BuildAnalyzerConfigurationInternal
 {
-    public BuildAnalyzerConfigurationInternal(string ruleId, EvaluationAnalysisScope evaluationAnalysisScope, BuildAnalyzerResultSeverity severity, bool isEnabled)
+    public BuildAnalyzerConfigurationInternal(string ruleId, EvaluationAnalysisScope evaluationAnalysisScope, BuildAnalyzerResultSeverity severity)
     {
+        if (severity == BuildAnalyzerResultSeverity.Default)
+        {
+            throw new ArgumentOutOfRangeException(nameof(severity), severity, "Severity 'Default' is not recognized by the BuilcCheck reports infrastructure");
```

Comment:
@JanKrivanek @baronfel FYI this situation could appear when the rule is not configured in the editorconfig however the default severity of authored check is set to default.

Reply:
tipo 'BuilcCheck'

Reply:
Btw. this should actually _never_ happen - as we call the ctor after merging 3 config sources - editorconfig, default configr from buildcheck, and the hardcoded default config - which has the severity of None:

https://github.com/dotnet/msbuild/pull/10330/files#diff-889c51f15e7c4415667244aaecf9a9c77442af8baefec35fe30a95ac3425915cR21-R25

But it's very fine to keep the exception here - to document that 'Default' seveirty will never be used in the effective configuration

Reply:
It should not of course, in terms of running the BuildCheck, however if we author the Check Rule with default Severity set to 'Default' and it was not configured, then we will raise the exception